### PR TITLE
Replace Travis with GitHub Actions

### PR DESCRIPTION
**Pull request by matyasselmeci**
https://crt.cs.wisc.edu/rt/Ticket/Display.html?id=102007

### CODE REVIEW

File: .github/workflows/build_deploy.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+name: Build/deploy web pages
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+jobs:
+  build-and-maybe-deploy:
```
Comment (brianhlin) [inline .github/workflows/build_deploy.yml:8]:
```suggestion
  build-and-maybe-deploy:
```

lol this name

File: .github/workflows/build_deploy.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: 2.6
+          bundler-cache: true
```
Comment (brianhlin) [inline .github/workflows/build_deploy.yml:15]: I haven't looked at Ruby stuff in a long while. Does this use the repo's [Gemfile](https://github.com/CHTC/chtc-website-source/blob/master/Gemfile) and lock the Jekyll version by virtue of the locked Ruby version?

File: script/cideploy [hunk 3]
```diff
@@ -2,48 +2,42 @@
...
-    git commit -F "$commitmessage"
-    if [[ -z $NO_PUSH ]]; then
-        git push -fq origin $BRANCH > /dev/null
-    fi
-    echo -e "Deploy completed\n"
+echo -e "Starting to deploy to Github Pages\n"
+
+if [[ $GITHUB_ACTIONS == true ]]; then
+    git config --global user.email "actions@github.com"
+    git config --global user.name "GitHub Actions"
```
Comment (brianhlin) [inline script/cideploy:22]: Why set the email to one we don't control? I can understand setting the name like that but I'm not sure I get why we'd want to set the email this way.

**Review by brianhlin — APPROVED** [review]
Some minor comments/questions but otherwise this looks like it should work.

Any reason why you're not switching this over to the Jekyll action (https://github.com/marketplace/actions/jekyll-actions)?

File: .github/workflows/build_deploy.yml (hunk 2 above)
Comment (matyasselmeci) [reply .github/workflows/build_deploy.yml:15]: I have no idea. This is basically the equivalent of the
```
rvm:
  - 2.6.3
```
block from travis.

File: script/cideploy (hunk 3 above)
Comment (matyasselmeci) [reply script/cideploy:22]: We need to set the email to _something_ or git won't let us commit. For the travis build we set it to "travis@travis-ci.org". If you have another idea, I can use that.

**Comment by matyasselmeci** [comment]
I'm not using the Jekyll action because I just wanted a straightforward translation of the current Travis build. I have no idea how the Jekyll action deals with things like our PHP web form or renaming `*.html` to `*.shtml` so getting it to work would require some adaptation and more testing. If we want, we can consider it later when I have more time...

File: .github/workflows/build_deploy.yml (hunk 2 above)
Comment (brianhlin) [reply .github/workflows/build_deploy.yml:15]: We should consider version locking Jekyll in the `Gemfile` then to be sure

File: script/cideploy (hunk 3 above)
Comment (brianhlin) [reply script/cideploy:22]: Mmk, that's fine by me then. Though I do wonder how `mkdocs` does it https://github.com/opensciencegrid/docs/commit/827b4d851ab659bc231a16fc0569900a25298a9b.

**Comment by brianhlin** [comment]
Sure, I've approved so merge whenever you're ready

**Pull request merged**